app/src/forms/form/service.js

```diff
@@ -274,6 +274,10 @@ const service = {
       else if (!obj.hidden && obj.components && obj.components.length) {
         fields.push(obj.components.flatMap(o => findFields(o)));
       }
+      // check for components inside columns
+      else if (!obj.hidden && obj.columns && obj.columns.length) {
+        fields.push(obj.columns.flatMap(o => findFields(o)));
+      }
```

Comment:
It may be better to consider rewriting the findFields function definition to be more generic with its recursion approach by checking all keys instead of only a few hard-coded ones like components and columns:
```js
    // Recursively find all field key names
    // TODO: Consider if this should be a form utils function instead?
    const findFields = (obj) => {
      const fields = [];
      if (!obj.hidden) {
        // Only add key if it is an input and visible
        if (obj.input) fields.push(obj.key);
        // Recursively check all children attributes that are arrays
        else {
          Object.keys(obj).forEach(key => {
            if (Array.isArray(obj[key]) && obj[key].length) {
              fields.push(obj[key].flatMap(o => findFields(o)));
            }
          });
        }
      }
      return fields.flat();
    };
```

Reply:
good idea. thanks for the code.